src/locales/scripts/read-i18n.js

```diff
@@ -0,0 +1,164 @@
+    this.children.push(child)
+  }
+
+  /**
+   * Get the JSON representation of this entry and all its children. This
+   * conversion loses information present in comments.
+   *
+   * @return {SimJson} a POJSO containing the translation mappings
+   */
+  toJSON() {
+    // This is a string-string entry, will be handled by parent.
+    if (this.value) return {}
+
+    /** @type {SimJson} */
+    const pojo = {}
+    for (const child of this.children) {
+      if (child.value) pojo[child.key] = child.value
+      else pojo[child.key] = child.toJSON()
+    }
+    return pojo
+  }
+}
+
+/**
+ * Determine the key for the entry, which will be a string literal if quoted,
+ * or an identifier if not quoted.
+ *
+ * @param keyNode {import('@babel/types').StringLiteral|import('@babel/types').Identifier}
+ * @return {string} the text content of the key
+ */
+const parseKey = (keyNode) => {
+  if (keyNode === undefined) return ''
+  switch (keyNode.type) {
+    case 'StringLiteral':
+      return keyNode.value
+    case 'Identifier':
+      return keyNode.name
+  }
+}
+
+/**
+ * Parse a single or multi-line comment. If the comment is multi-line, newlines
+ * and asterisks will be removed from the output.
+ *
+ * @param commentNode {import('@babel/types').CommentLine|import('@babel/types').CommentBlock}
+ * @return {string} the text content of the comment
+ */
+const parseComment = (commentNode) => {
+  switch (commentNode.type) {
+    case 'CommentLine':
+      return commentNode.value.trim()
+    case 'CommentBlock':
+      return commentNode.value
+        .replace(/\n|\*+/g, '')
+        .replace(/\s+/, ' ')
+        .trim()
+  }
+}
+
+/**
+ * Populate the value or children of the `Entry` depending on whether the value
+ * is a string or an object expression.
+ *
+ * @param entry {Entry} the entry to populate
+ * @param valueNode {import('@babel/types').StringLiteral|import('@babel/types').ObjectExpression}
+ */
+const parseValue = (entry, valueNode) => {
+  switch (valueNode.type) {
+    case 'StringLiteral':
+      entry.value = valueNode.value
+      break
+    case 'ObjectExpression':
+      valueNode.properties.map(parseObjProperty).forEach((child) => {
+        entry.addChild(child)
+      })
+      break
+  }
+}
+
+/**
+ * Create an `Entry` instance by parsing the AST node from Babel.
+ *
+ * @param node {import('@babel/types').ObjectProperty}
+ * @return {Entry} the entry generated by parsing the node
+ */
+const parseObjProperty = (node) => {
+  let key = parseKey(node.key)
+  let comments = node.leadingComments?.map(parseComment).join('')
+  let entry = new Entry(key, comments)
+  parseValue(entry, node.value)
+  return entry
+}
+
+/**
+ * Parse the given filename into a tree of `Entry` instances.
+ * @param filename {string} the name of the JSON file to read
+ * @return {Entry} the root `Entry` instance for the top-level JSON object
+ */
+const parseJson = (filename) =>
+  parseObjProperty({
+    value: babel.parseExpression(
```

Comment:
Nice to use `parseExpression` here and avoid the hacky ones I suggested :sweat_smile: